spec/features/proposal/proposal_progress_spec.rb

```diff
@@ -1,131 +1,245 @@
+            expect(page).to have_css('div.timeline-node')
+            expect(page).to have_css('div.timeline-line')
+            expect(page).to have_content('Submitted: 2019-10-11 01:00')
+            expect(page).to have_content('By: TestUser1')
+          end
+
+          within '#timeline-review-submission' do
+            expect(page).to have_content('Approval')
+            expect(page).to have_content('Rejected: 2019-10-11 03:00')
+            expect(page).to have_content('By: TestUser3')
+            expect(page).to have_content('Reason: TestReason')
+            expect(page).to have_css('div.timeline-node-active')
+            expect(page).to have_css('div.timeline-faded-line-active')
+          end
+        end
+
+        it 'displays the correct actions' do
+          within '.progress-actions' do
+            expect(page).to have_content('You may rescind this proposal to make additional changes.')
+            expect(page).to have_link('Cancel Proposal Submission')
+          end
+        end
       end
     end
 
-    it 'displays the correct actions' do
-      within '.progress-actions' do
-        expect(page).to have_content('No actions are possible.')
+    context 'when viewing a done proposal' do
+      before do
+        mock_publish(@collection_draft_proposal)
+        visit progress_collection_draft_proposal_path(@collection_draft_proposal)
+      end
+
+      it 'displays the correct progress' do
+        within '#timeline-create-submission' do
+          expect(page).to have_content('Submitted for Review')
+          expect(page).to have_content('Submitted: 2019-10-11 01:00')
+          expect(page).to have_content('By: TestUser1')
+          expect(page).to have_css('div.timeline-node')
+          expect(page).to have_css('div.timeline-line')
+        end
+
+        within '#timeline-review-submission' do
+          expect(page).to have_content('Approval')
+          expect(page).to have_content('Approved: 2019-10-11 02:00')
+          expect(page).to have_content('By: TestUser2')
+          expect(page).to have_css('div.timeline-node')
+          expect(page).to have_css('div.timeline-line')
+        end
+
+        within '#timeline-waiting-to-publish' do
+          expect(page).to have_css('div.timeline-node')
+          expect(page).to have_css('div.timeline-line')
+        end
+
+        within '#timeline-published' do
+          expect(page).to have_css('div.timeline-node-active')
+          expect(page).to have_content('Published: 2019-10-11 04:00')
+          expect(page).to have_content('By: TestUser4')
+        end
+      end
+
+      it 'displays the correct actions' do
+        within '.progress-actions' do
+          expect(page).to have_content('No actions are possible.')
+        end
       end
     end
   end
 
-  context 'when viewing a rejected proposal' do
-    context 'when viewing a rescinded rejected proposal' do
+  context 'when viewing the progress page as an approver' do
```

Comment:
Every line from here to the end of the document is new, but the only things that really change are a small number of the expected text strings.